gum/backend-arm64/gumstalker-arm64.c

```diff
@@ -2570,6 +2579,15 @@ gum_exec_block_write_call_invoke_code (GumExecBlock * block,
   if (target->reg != ARM64_REG_INVALID)
   {
     entry_func = GUM_ENTRYGATE (call_reg);
+
+    gum_arm64_writer_put_call_address_with_arguments (cw,
+        GUM_ADDRESS (gum_stalker_is_address_included), 2,
+        GUM_ARG_ADDRESS, GUM_ADDRESS (block->ctx->stalker),
+        GUM_ARG_REGISTER, ARM64_REG_X15);
+
+    gum_arm64_writer_put_mov_reg_reg (cw, ARM64_REG_X15, ARM64_REG_X0);
+    gum_arm64_writer_put_cbz_reg_label (cw, ARM64_REG_X15, keep_this_blr);
```

Comment:
So if we change it to return a `gboolean` that will simplify this logic a bit.

Reply:
That was a trick (actually just to save an instruction). In the case of valid register not excluded, we need the target address in register X15. We should have done this by pushing and popping X15 around the `gum_stalker_is_address_included`call. With this solution we directly use the return value of the function by moving X0 into X15, that is 0 (false) when the address need to be excluded.
I know it's micro-optimization, but if you think it's clearer use a gboolean return value, I'll change it.